In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# malaria_2003_df = pd.read_csv('malaria_notification_data_csv/NOTIPO03.csv')
# malaria_2004_df = pd.read_csv('malaria_notification_data_csv/NOTIPO04.csv')
# malaria_2005_df = pd.read_csv('malaria_notification_data_csv/NOTIPO05.csv')
# malaria_2006_df = pd.read_csv('malaria_notification_data_csv/NOTIPO06.csv')
# malaria_2007_df = pd.read_csv('malaria_notification_data_csv/NOTIPO07.csv')
# malaria_2008_df = pd.read_csv('malaria_notification_data_csv/NOTIPO08.csv')
# malaria_2009_df = pd.read_csv('malaria_notification_data_csv/NOTIPO09.csv')
# malaria_2010_df = pd.read_csv('malaria_notification_data_csv/NOTIPO10.csv')
# malaria_2011_df = pd.read_csv('malaria_notification_data_csv/NOTIPO11.csv')
# malaria_2012_df = pd.read_csv('malaria_notification_data_csv/NOTIPO12.csv')
# malaria_2013_df = pd.read_csv('malaria_notification_data_csv/NOTIPO13.csv')
# malaria_2014_df = pd.read_csv('malaria_notification_data_csv/NOTIPO14.csv')
# malaria_2015_df = pd.read_csv('malaria_notification_data_csv/NOTIPO15.csv')
# malaria_2016_df = pd.read_csv('malaria_notification_data_csv/NOTIPO16.csv')
malaria_2017_df = pd.read_csv('malaria_notification_data_csv/NOTIPO17.csv')
malaria_2018_df = pd.read_csv('malaria_notification_data_csv/NOTIPO18.csv')
malaria_2019_df = pd.read_csv('malaria_notification_data_csv/NOTIPO19.csv')
malaria_2020_df = pd.read_csv('malaria_notification_data_csv/NOTIPO20.csv')
malaria_2021_df = pd.read_csv('malaria_notification_data_csv/NOTIPO21.csv')
# malaria_2022_df = pd.read_csv('malaria_notification_data_csv/NOTIPO22.csv')
# malaria_2023_df = pd.read_csv('malaria_notification_data_csv/NOTIPO23.csv')

C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (37,38,41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (29,30,41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (30) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_na

In [3]:
malaria_2017_2021_df = pd.concat([malaria_2017_df, malaria_2018_df, malaria_2019_df, malaria_2020_df, malaria_2021_df])

malaria_2017_2021_df = malaria_2017_2021_df.reset_index().drop(columns=['index'])
malaria_2017_2021_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1   
...           ...       ...         ...       ...      ...       ...   
1006355       139  19/07/21  20/07/2021  20/07/21  29/2021         1   
1006356      3156  07/07/21  14/07/2021  14/07/21  27/2021         1   
1006357       697  16/08/21  25/08/2021  25/08/21  33/2021         1   
1006358      4779  21/07/21  22/07/2021  22/07/21  29/2021         1   
1006359       571  09/07/21  30/07/2021  12/07/21  27/2021         1   

         UF_NOTIF  MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  \
0              13    130260       137    4274.0  ...                  NaN   
1              13    130260       137    2118.0  ...                  NaN   
2              13    130260       137    4274.0  ...                  NaN   
3              13    130260       137    4060.0  ...                  NaN   
4              13    130260        35    3601.0  ...                  NaN   
...           ...       ...       ...       ...  ...                  ...   
1006355        16    160050       128  339967.0  ...                  NaN   
1006356        11    110020         3   10424.0  ...                  NaN   
1006357        13    130353         9      31.0  ...                  NaN   
1006358        13    130380        48     666.0  ...                  NaN   
1006359        13    130250        68      19.0  ...                  NaN   

         EXAMINADOR ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  \
0            4315.0       2         6          NaN           NaN   
1            2118.0       2         6          NaN           NaN   
2            4315.0       1         5          NaN           NaN   
3              17.0       2         6          NaN           NaN   
4            3601.0       2         6          NaN           NaN   
...             ...     ...       ...          ...           ...   
1006355     95384.0       2         6          NaN           NaN   
1006356     10424.0       2         6          NaN           NaN   
1006357        31.0       1         6          NaN           NaN   
1006358       135.0       2         6          NaN           NaN   
1006359        19.0       1         6          NaN           NaN   

         DIMEA_AMAMENT  VIVAX  FALCIPARUM  NIV_ESCO  
0                  NaN    2.0         2.0       5.0  
1                  NaN    2.0         2.0       4.0  
2                  NaN    1.0         2.0       6.0  
3                  NaN    2.0         2.0       3.0  
4                  NaN    2.0         2.0       5.0  
...                ...    ...         ...       ...  
1006355            NaN    2.0         2.0       4.0  
1006356            NaN    2.0         2.0       6.0  
1006357            NaN    1.0         2.0       3.0  
1006358            NaN    2.0         2.0       5.0  
1006359            NaN    1.0         2.0       4.0  

[1006360 rows x 48 columns]

#### Gets vivax cases (RES_EXAM = 4)

In [4]:
vivax_cases_df = malaria_2017_2021_df[malaria_2017_2021_df.get('RES_EXAM')==4]
vivax_cases_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1   
...           ...       ...         ...       ...      ...       ...   
1006354      1639  10/09/21  05/01/2022  05/01/22  36/2021         2   
1006355       139  19/07/21  20/07/2021  20/07/21  29/2021         1   
1006357       697  16/08/21  25/08/2021  25/08/21  33/2021         1   
1006358      4779  21/07/21  22/07/2021  22/07/21  29/2021         1   
1006359       571  09/07/21  30/07/2021  12/07/21  27/2021         1   

         UF_NOTIF  MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  \
0              13    130260       137    4274.0  ...                  NaN   
1              13    130260       137    2118.0  ...                  NaN   
2              13    130260       137    4274.0  ...                  NaN   
3              13    130260       137    4060.0  ...                  NaN   
4              13    130260        35    3601.0  ...                  NaN   
...           ...       ...       ...       ...  ...                  ...   
1006354        14    140002       249     320.0  ...                  NaN   
1006355        16    160050       128  339967.0  ...                  NaN   
1006357        13    130353         9      31.0  ...                  NaN   
1006358        13    130380        48     666.0  ...                  NaN   
1006359        13    130250        68      19.0  ...                  NaN   

         EXAMINADOR ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  \
0            4315.0       2         6          NaN           NaN   
1            2118.0       2         6          NaN           NaN   
2            4315.0       1         5          NaN           NaN   
3              17.0       2         6          NaN           NaN   
4            3601.0       2         6          NaN           NaN   
...             ...     ...       ...          ...           ...   
1006354       320.0       2         6          NaN           NaN   
1006355     95384.0       2         6          NaN           NaN   
1006357        31.0       1         6          NaN           NaN   
1006358       135.0       2         6          NaN           NaN   
1006359        19.0       1         6          NaN           NaN   

         DIMEA_AMAMENT  VIVAX  FALCIPARUM  NIV_ESCO  
0                  NaN    2.0         2.0       5.0  
1                  NaN    2.0         2.0       4.0  
2                  NaN    1.0         2.0       6.0  
3                  NaN    2.0         2.0       3.0  
4                  NaN    2.0         2.0       5.0  
...                ...    ...         ...       ...  
1006354            NaN    2.0         2.0       NaN  
1006355            NaN    2.0         2.0       4.0  
1006357            NaN    1.0         2.0       3.0  
1006358            NaN    2.0         2.0       5.0  
1006359            NaN    1.0         2.0       4.0  

[884532 rows x 48 columns]

In [5]:
len(vivax_cases_df)/len(malaria_2017_2021_df)*100

87.89419293294645

About 88% of all malaria cases between 2017 and 2021 were caused by P. Vivax

#### Gets name of UFs

In [6]:
uf_dict = {11 : 'RO',
           12 : 'AC',
           13 : 'AM',
           14 : 'RR',
           15 : 'PA',
           16 : 'AP',
           17 : 'TO',
           21 : 'MA',
           22 : 'PI',
           23 : 'CE',
           24 : 'RN',
           25 : 'PB',
           26 : 'PE',
           27 : 'AL',
           28 : 'SE',
           29 : 'BA',
           31 : 'MG', 
           32 : 'ES',
           33 : 'RJ',
           35 : 'SP',
           41 : 'PR',
           42 : 'SC',
           43 : 'RS',
           50 : 'MS',
           51 : 'MT',
           52 : 'GO',
           53 : 'DF'
          }
corrected_ufs_df = vivax_cases_df.replace({"UF_NOTIF": uf_dict, "UF_RESID": uf_dict})
corrected_ufs_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1006354      1639  10/09/21  05/01/2022  05/01/22  36/2021         2       RR   
1006355       139  19/07/21  20/07/2021  20/07/21  29/2021         1       AP   
1006357       697  16/08/21  25/08/2021  25/08/21  33/2021         1       AM   
1006358      4779  21/07/21  22/07/2021  22/07/21  29/2021         1       AM   
1006359       571  09/07/21  30/07/2021  12/07/21  27/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          130260       137    4274.0  ...                  NaN      4315.0   
1          130260       137    2118.0  ...                  NaN      2118.0   
2          130260       137    4274.0  ...                  NaN      4315.0   
3          130260       137    4060.0  ...                  NaN        17.0   
4          130260        35    3601.0  ...                  NaN      3601.0   
...           ...       ...       ...  ...                  ...         ...   
1006354    140002       249     320.0  ...                  NaN       320.0   
1006355    160050       128  339967.0  ...                  NaN     95384.0   
1006357    130353         9      31.0  ...                  NaN        31.0   
1006358    130380        48     666.0  ...                  NaN       135.0   
1006359    130250        68      19.0  ...                  NaN        19.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             2         6          NaN           NaN            NaN   2.0   
1             2         6          NaN           NaN            NaN   2.0   
2             1         5          NaN           NaN            NaN   1.0   
3             2         6          NaN           NaN            NaN   2.0   
4             2         6          NaN           NaN            NaN   2.0   
...         ...       ...          ...           ...            ...   ...   
1006354       2         6          NaN           NaN            NaN   2.0   
1006355       2         6          NaN           NaN            NaN   2.0   
1006357       1         6          NaN           NaN            NaN   1.0   
1006358       2         6          NaN           NaN            NaN   2.0   
1006359       1         6          NaN           NaN            NaN   1.0   

         FALCIPARUM  NIV_ESCO  
0               2.0       5.0  
1               2.0       4.0  
2               2.0       6.0  
3               2.0       3.0  
4               2.0       5.0  
...             ...       ...  
1006354         2.0       NaN  
1006355         2.0       4.0  
1006357         2.0       3.0  
1006358         2.0       5.0  
1006359         2.0       4.0  

[884532 rows x 48 columns]

#### Gets data for the Amazon region

In [7]:
amazon_df = corrected_ufs_df[corrected_ufs_df.get('UF_NOTIF').isin(['RO', 'AC', 'AM', 'RR', 'PA', 'AP', 'TO'])]
amazon_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1006354      1639  10/09/21  05/01/2022  05/01/22  36/2021         2       RR   
1006355       139  19/07/21  20/07/2021  20/07/21  29/2021         1       AP   
1006357       697  16/08/21  25/08/2021  25/08/21  33/2021         1       AM   
1006358      4779  21/07/21  22/07/2021  22/07/21  29/2021         1       AM   
1006359       571  09/07/21  30/07/2021  12/07/21  27/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          130260       137    4274.0  ...                  NaN      4315.0   
1          130260       137    2118.0  ...                  NaN      2118.0   
2          130260       137    4274.0  ...                  NaN      4315.0   
3          130260       137    4060.0  ...                  NaN        17.0   
4          130260        35    3601.0  ...                  NaN      3601.0   
...           ...       ...       ...  ...                  ...         ...   
1006354    140002       249     320.0  ...                  NaN       320.0   
1006355    160050       128  339967.0  ...                  NaN     95384.0   
1006357    130353         9      31.0  ...                  NaN        31.0   
1006358    130380        48     666.0  ...                  NaN       135.0   
1006359    130250        68      19.0  ...                  NaN        19.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             2         6          NaN           NaN            NaN   2.0   
1             2         6          NaN           NaN            NaN   2.0   
2             1         5          NaN           NaN            NaN   1.0   
3             2         6          NaN           NaN            NaN   2.0   
4             2         6          NaN           NaN            NaN   2.0   
...         ...       ...          ...           ...            ...   ...   
1006354       2         6          NaN           NaN            NaN   2.0   
1006355       2         6          NaN           NaN            NaN   2.0   
1006357       1         6          NaN           NaN            NaN   1.0   
1006358       2         6          NaN           NaN            NaN   2.0   
1006359       1         6          NaN           NaN            NaN   1.0   

         FALCIPARUM  NIV_ESCO  
0               2.0       5.0  
1               2.0       4.0  
2               2.0       6.0  
3               2.0       3.0  
4               2.0       5.0  
...             ...       ...  
1006354         2.0       NaN  
1006355         2.0       4.0  
1006357         2.0       3.0  
1006358         2.0       5.0  
1006359         2.0       4.0  

[866720 rows x 48 columns]

In [8]:
amazonas_df = amazon_df[amazon_df.get('UF_NOTIF')=='AM']
amazonas_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1006350        39  03/05/21  25/02/2022  13/08/21  18/2021         1       AM   
1006353      1350  29/11/21  09/12/2021  09/12/21  48/2021         1       AM   
1006357       697  16/08/21  25/08/2021  25/08/21  33/2021         1       AM   
1006358      4779  21/07/21  22/07/2021  22/07/21  29/2021         1       AM   
1006359       571  09/07/21  30/07/2021  12/07/21  27/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          130260       137    4274.0  ...                  NaN      4315.0   
1          130260       137    2118.0  ...                  NaN      2118.0   
2          130260       137    4274.0  ...                  NaN      4315.0   
3          130260       137    4060.0  ...                  NaN        17.0   
4          130260        35    3601.0  ...                  NaN      3601.0   
...           ...       ...       ...  ...                  ...         ...   
1006350    130340         1     389.0  ...                  NaN       389.0   
1006353    130006        11      95.0  ...                  NaN        95.0   
1006357    130353         9      31.0  ...                  NaN        31.0   
1006358    130380        48     666.0  ...                  NaN       135.0   
1006359    130250        68      19.0  ...                  NaN        19.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             2         6          NaN           NaN            NaN   2.0   
1             2         6          NaN           NaN            NaN   2.0   
2             1         5          NaN           NaN            NaN   1.0   
3             2         6          NaN           NaN            NaN   2.0   
4             2         6          NaN           NaN            NaN   2.0   
...         ...       ...          ...           ...            ...   ...   
1006350       2         6          NaN           NaN            NaN   2.0   
1006353       2         6          NaN           NaN            NaN   2.0   
1006357       1         6          NaN           NaN            NaN   1.0   
1006358       2         6          NaN           NaN            NaN   2.0   
1006359       1         6          NaN           NaN            NaN   1.0   

         FALCIPARUM  NIV_ESCO  
0               2.0       5.0  
1               2.0       4.0  
2               2.0       6.0  
3               2.0       3.0  
4               2.0       5.0  
...             ...       ...  
1006350         2.0       4.0  
1006353         2.0       6.0  
1006357         2.0       3.0  
1006358         2.0       5.0  
1006359         2.0       4.0  

[335206 rows x 48 columns]

In [9]:
len(amazonas_df)/len(vivax_cases_df)*100

37.89642432382322

Around 37.9% of all malaria cases caused by P. Vivax between 2017 and 2021 were notified in the state of Amazonas

#### Cases notified of Amazonas' residents in the state 

In [10]:
amazonas_notif_and_resid = amazon_df[(amazon_df.get('UF_NOTIF')=='AM') & (amazon_df.get('UF_RESID')=='AM')]
amazonas_notif_and_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1006338       258  20/09/21  04/10/2021  30/09/21  38/2021         2       AM   
1006350        39  03/05/21  25/02/2022  13/08/21  18/2021         1       AM   
1006357       697  16/08/21  25/08/2021  25/08/21  33/2021         1       AM   
1006358      4779  21/07/21  22/07/2021  22/07/21  29/2021         1       AM   
1006359       571  09/07/21  30/07/2021  12/07/21  27/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          130260       137    4274.0  ...                  NaN      4315.0   
1          130260       137    2118.0  ...                  NaN      2118.0   
2          130260       137    4274.0  ...                  NaN      4315.0   
3          130260       137    4060.0  ...                  NaN        17.0   
4          130260        35    3601.0  ...                  NaN      3601.0   
...           ...       ...       ...  ...                  ...         ...   
1006338    130080        65     132.0  ...                  NaN        84.0   
1006350    130340         1     389.0  ...                  NaN       389.0   
1006357    130353         9      31.0  ...                  NaN        31.0   
1006358    130380        48     666.0  ...                  NaN       135.0   
1006359    130250        68      19.0  ...                  NaN        19.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             2         6          NaN           NaN            NaN   2.0   
1             2         6          NaN           NaN            NaN   2.0   
2             1         5          NaN           NaN            NaN   1.0   
3             2         6          NaN           NaN            NaN   2.0   
4             2         6          NaN           NaN            NaN   2.0   
...         ...       ...          ...           ...            ...   ...   
1006338       2         6          NaN           NaN            NaN   2.0   
1006350       2         6          NaN           NaN            NaN   2.0   
1006357       1         6          NaN           NaN            NaN   1.0   
1006358       2         6          NaN           NaN            NaN   2.0   
1006359       1         6          NaN           NaN            NaN   1.0   

         FALCIPARUM  NIV_ESCO  
0               2.0       5.0  
1               2.0       4.0  
2               2.0       6.0  
3               2.0       3.0  
4               2.0       5.0  
...             ...       ...  
1006338         2.0       5.0  
1006350         2.0       4.0  
1006357         2.0       3.0  
1006358         2.0       5.0  
1006359         2.0       4.0  

[333650 rows x 48 columns]

In [11]:
len(amazonas_notif_and_resid)/len(amazonas_df)*100

99.53580783160206

From all cases notified in the state of Amazonas, about 99.5% of the infected were residents of the state.

#### Cases registered on Amazonas based on municipality of residence

In [12]:
cases_by_AM_municipality = amazonas_notif_and_resid[['MUN_RESI']].groupby(['MUN_RESI']).size().reset_index(name='TOTAL_CASES').sort_values(by='TOTAL_CASES', ascending=False)
cases_by_AM_municipality['MUN_RESI'] = cases_by_AM_municipality['MUN_RESI'].astype(int)
cases_by_AM_municipality

MUN_RESI  TOTAL_CASES
37    130260        51965
51    130380        47145
7     130040        35446
49    130360        17129
19    130120        16392
..       ...          ...
1     130006           28
2     130008           23
58    130423           17
3     130010           13
61    130440           13

[62 rows x 2 columns]

It is possible to see that Manaus is the city of residence of most notified cases in the state of Amazonas

#### Cases notified of Manaus' residents in the city 

In [13]:
manaus_notif_and_resid = corrected_ufs_df[(corrected_ufs_df.get('MUN_NOTI')==130260) & (corrected_ufs_df.get('MUN_RESI')==130260)]
manaus_notif_and_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1005991      2653  25/09/21  29/09/2021  28/09/21  38/2021         1       AM   
1006093       181  03/11/21  04/11/2021  04/11/21  44/2021         1       AM   
1006111    187216  08/10/21  10/10/2021  08/10/21  40/2021         1       AM   
1006159       148  19/10/21  21/10/2021  21/10/21  42/2021         1       AM   
1006319        67  02/08/21  03/08/2021  03/08/21  31/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          130260       137    4274.0  ...                  NaN      4315.0   
1          130260       137    2118.0  ...                  NaN      2118.0   
2          130260       137    4274.0  ...                  NaN      4315.0   
3          130260       137    4060.0  ...                  NaN        17.0   
4          130260        35    3601.0  ...                  NaN      3601.0   
...           ...       ...       ...  ...                  ...         ...   
1005991    130260       411    2573.0  ...                 14,5      2573.0   
1006093    130260       385    4799.0  ...                   11      3208.0   
1006111    130260        70     338.0  ...                 13,8      2056.0   
1006159    130260        42    3168.0  ...                  NaN       593.0   
1006319    130260        46    2712.0  ...                  NaN      3278.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             2         6          NaN           NaN            NaN   2.0   
1             2         6          NaN           NaN            NaN   2.0   
2             1         5          NaN           NaN            NaN   1.0   
3             2         6          NaN           NaN            NaN   2.0   
4             2         6          NaN           NaN            NaN   2.0   
...         ...       ...          ...           ...            ...   ...   
1005991       2         6          NaN           NaN            NaN   2.0   
1006093       2         5          2.0           NaN            NaN   2.0   
1006111       2         6          NaN           NaN            NaN   2.0   
1006159       2         6          NaN           NaN            NaN   2.0   
1006319       2         5          2.0           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
0               2.0       5.0  
1               2.0       4.0  
2               2.0       6.0  
3               2.0       3.0  
4               2.0       5.0  
...             ...       ...  
1005991         2.0       5.0  
1006093         2.0       6.0  
1006111         2.0       5.0  
1006159         2.0       3.0  
1006319         2.0       5.0  

[51550 rows x 48 columns]

#### Cases notified in Manaus

In [14]:
manaus_notif = corrected_ufs_df[corrected_ufs_df.get('MUN_NOTI')==130260]
manaus_notif

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1005999    188515  27/12/21  03/01/2022  27/12/21  52/2021         1       AM   
1006093       181  03/11/21  04/11/2021  04/11/21  44/2021         1       AM   
1006111    187216  08/10/21  10/10/2021  08/10/21  40/2021         1       AM   
1006159       148  19/10/21  21/10/2021  21/10/21  42/2021         1       AM   
1006319        67  02/08/21  03/08/2021  03/08/21  31/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          130260       137    4274.0  ...                  NaN      4315.0   
1          130260       137    2118.0  ...                  NaN      2118.0   
2          130260       137    4274.0  ...                  NaN      4315.0   
3          130260       137    4060.0  ...                  NaN        17.0   
4          130260        35    3601.0  ...                  NaN      3601.0   
...           ...       ...       ...  ...                  ...         ...   
1005999    130260        70     340.0  ...                 12,8       430.0   
1006093    130260       385    4799.0  ...                   11      3208.0   
1006111    130260        70     338.0  ...                 13,8      2056.0   
1006159    130260        42    3168.0  ...                  NaN       593.0   
1006319    130260        46    2712.0  ...                  NaN      3278.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             2         6          NaN           NaN            NaN   2.0   
1             2         6          NaN           NaN            NaN   2.0   
2             1         5          NaN           NaN            NaN   1.0   
3             2         6          NaN           NaN            NaN   2.0   
4             2         6          NaN           NaN            NaN   2.0   
...         ...       ...          ...           ...            ...   ...   
1005999       2         5          2.0           NaN            NaN   2.0   
1006093       2         5          2.0           NaN            NaN   2.0   
1006111       2         6          NaN           NaN            NaN   2.0   
1006159       2         6          NaN           NaN            NaN   2.0   
1006319       2         5          2.0           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
0               2.0       5.0  
1               2.0       4.0  
2               2.0       6.0  
3               2.0       3.0  
4               2.0       5.0  
...             ...       ...  
1005999         2.0       4.0  
1006093         2.0       6.0  
1006111         2.0       5.0  
1006159         2.0       3.0  
1006319         2.0       5.0  

[52549 rows x 48 columns]

#### Cases of residents of Manaus

In [15]:
manaus_resid = corrected_ufs_df[corrected_ufs_df.get('MUN_RESI')==130260]
manaus_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1005991      2653  25/09/21  29/09/2021  28/09/21  38/2021         1       AM   
1006093       181  03/11/21  04/11/2021  04/11/21  44/2021         1       AM   
1006111    187216  08/10/21  10/10/2021  08/10/21  40/2021         1       AM   
1006159       148  19/10/21  21/10/2021  21/10/21  42/2021         1       AM   
1006319        67  02/08/21  03/08/2021  03/08/21  31/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          130260       137    4274.0  ...                  NaN      4315.0   
1          130260       137    2118.0  ...                  NaN      2118.0   
2          130260       137    4274.0  ...                  NaN      4315.0   
3          130260       137    4060.0  ...                  NaN        17.0   
4          130260        35    3601.0  ...                  NaN      3601.0   
...           ...       ...       ...  ...                  ...         ...   
1005991    130260       411    2573.0  ...                 14,5      2573.0   
1006093    130260       385    4799.0  ...                   11      3208.0   
1006111    130260        70     338.0  ...                 13,8      2056.0   
1006159    130260        42    3168.0  ...                  NaN       593.0   
1006319    130260        46    2712.0  ...                  NaN      3278.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             2         6          NaN           NaN            NaN   2.0   
1             2         6          NaN           NaN            NaN   2.0   
2             1         5          NaN           NaN            NaN   1.0   
3             2         6          NaN           NaN            NaN   2.0   
4             2         6          NaN           NaN            NaN   2.0   
...         ...       ...          ...           ...            ...   ...   
1005991       2         6          NaN           NaN            NaN   2.0   
1006093       2         5          2.0           NaN            NaN   2.0   
1006111       2         6          NaN           NaN            NaN   2.0   
1006159       2         6          NaN           NaN            NaN   2.0   
1006319       2         5          2.0           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
0               2.0       5.0  
1               2.0       4.0  
2               2.0       6.0  
3               2.0       3.0  
4               2.0       5.0  
...             ...       ...  
1005991         2.0       5.0  
1006093         2.0       6.0  
1006111         2.0       5.0  
1006159         2.0       3.0  
1006319         2.0       5.0  

[52079 rows x 48 columns]

#### Year of registration of notification based on residents of Manaus

In [16]:
manaus_resid['YEAR_NOTIF'] = manaus_resid['DT_NOTIF'].str[-2:].astype(int)
manaus_resid['YEAR_NOTIF'] = manaus_resid['YEAR_NOTIF'].apply(lambda x: 2000 + x if x <= 23 else 1900 + x)
manaus_resid

<ipython-input-16-bb7fdcbcbd17>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  manaus_resid['YEAR_NOTIF'] = manaus_resid['DT_NOTIF'].str[-2:].astype(int)
<ipython-input-16-bb7fdcbcbd17>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  manaus_resid['YEAR_NOTIF'] = manaus_resid['YEAR_NOTIF'].apply(lambda x: 2000 + x if x <= 23 else 1900 + x)


COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1005991      2653  25/09/21  29/09/2021  28/09/21  38/2021         1       AM   
1006093       181  03/11/21  04/11/2021  04/11/21  44/2021         1       AM   
1006111    187216  08/10/21  10/10/2021  08/10/21  40/2021         1       AM   
1006159       148  19/10/21  21/10/2021  21/10/21  42/2021         1       AM   
1006319        67  02/08/21  03/08/2021  03/08/21  31/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... EXAMINADOR  ID_LVC  GESTANTE   \
0          130260       137    4274.0  ...     4315.0        2         6   
1          130260       137    2118.0  ...     2118.0        2         6   
2          130260       137    4274.0  ...     4315.0        1         5   
3          130260       137    4060.0  ...       17.0        2         6   
4          130260        35    3601.0  ...     3601.0        2         6   
...           ...       ...       ...  ...        ...      ...       ...   
1005991    130260       411    2573.0  ...     2573.0        2         6   
1006093    130260       385    4799.0  ...     3208.0        2         5   
1006111    130260        70     338.0  ...     2056.0        2         6   
1006159    130260        42    3168.0  ...      593.0        2         6   
1006319    130260        46    2712.0  ...     3278.0        2         5   

        AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT  VIVAX FALCIPARUM  NIV_ESCO  \
0               NaN           NaN            NaN    2.0        2.0       5.0   
1               NaN           NaN            NaN    2.0        2.0       4.0   
2               NaN           NaN            NaN    1.0        2.0       6.0   
3               NaN           NaN            NaN    2.0        2.0       3.0   
4               NaN           NaN            NaN    2.0        2.0       5.0   
...             ...           ...            ...    ...        ...       ...   
1005991         NaN           NaN            NaN    2.0        2.0       5.0   
1006093         2.0           NaN            NaN    2.0        2.0       6.0   
1006111         NaN           NaN            NaN    2.0        2.0       5.0   
1006159         NaN           NaN            NaN    2.0        2.0       3.0   
1006319         2.0           NaN            NaN    2.0        2.0       5.0   

         YEAR_NOTIF  
0              2017  
1              2017  
2              2017  
3              2017  
4              2017  
...             ...  
1005991        2021  
1006093        2021  
1006111        2021  
1006159        2021  
1006319        2021  

[52079 rows x 49 columns]

#### Total cases of infected residents of Manaus by year 

In [17]:
cases_manaus_by_year = manaus_resid[['YEAR_NOTIF']].groupby(['YEAR_NOTIF']).size().reset_index(name='TOTAL_CASES').sort_values(by='TOTAL_CASES', ascending=False)
cases_manaus_by_year['YEAR_NOTIF'] = cases_manaus_by_year['YEAR_NOTIF'].astype(int)
cases_manaus_by_year

YEAR_NOTIF  TOTAL_CASES
0        2017        16042
1        2018        12618
2        2019         9720
3        2020         7501
4        2021         6198

#### Gets population data for all brazillian municipalities from 2003 to 2023

In [18]:
pop_df = pd.read_csv('pop_data/pop_2003-2023.csv')
pop_df = pop_df.drop(columns=['Unnamed: 0'])
pop_df

MUNIC_RES   ANO  POPULACAO
0          110001  2003      27563
1          110002  2003      79680
2          110003  2003       7478
3          110004  2003      74758
4          110005  2003      17794
...           ...   ...        ...
105771    5222005  2021      14088
105772    5222054  2021       9002
105773    5222203  2021       6451
105774    5222302  2021       5941
105775    5300108  2021    3094325

[105776 rows x 3 columns]

In [19]:
full_pop_df = pop_df.rename(columns={'ANO': 'YEAR', 'POPULACAO': 'POPULATION'})
full_pop_df = full_pop_df[full_pop_df.get('YEAR') >= 2017]
full_pop_df

MUNIC_RES  YEAR  POPULATION
77926     1100015  2017       25437
77927     1100023  2017      107345
77928     1100031  2017        6224
77929     1100049  2017       88507
77930     1100056  2017       17934
...           ...   ...         ...
105771    5222005  2021       14088
105772    5222054  2021        9002
105773    5222203  2021        6451
105774    5222302  2021        5941
105775    5300108  2021     3094325

[27850 rows x 3 columns]

#### Total cases by municipality of residence 

In [20]:
tot_cases_brazil = corrected_ufs_df[['MUN_RESI']].groupby(['MUN_RESI']).size().reset_index(name='TOTAL_CASES').sort_values(by='TOTAL_CASES', ascending=False)
tot_cases_brazil['MUN_RESI'] = tot_cases_brazil['MUN_RESI'].astype(int)
tot_cases_brazil['MUN_RESI'] = tot_cases_brazil['MUN_RESI'].astype(str)
tot_cases_brazil

MUN_RESI  TOTAL_CASES
111   130260        52079
57    120020        48668
125   130380        47146
81    130040        35567
16    110020        30664
..       ...          ...
565   351907            1
566   351930            1
567   352690            1
568   352730            1
738   539904            1

[739 rows x 2 columns]

Adding an year column for date of notification

In [21]:
corrected_ufs_df['YEAR_NOTIF'] = corrected_ufs_df['DT_NOTIF'].str[-2:].astype(int)
corrected_ufs_df['YEAR_NOTIF'] = corrected_ufs_df['YEAR_NOTIF'].apply(lambda x: 2000 + x if x <= 23 else 1900 + x)
corrected_ufs_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1006354      1639  10/09/21  05/01/2022  05/01/22  36/2021         2       RR   
1006355       139  19/07/21  20/07/2021  20/07/21  29/2021         1       AP   
1006357       697  16/08/21  25/08/2021  25/08/21  33/2021         1       AM   
1006358      4779  21/07/21  22/07/2021  22/07/21  29/2021         1       AM   
1006359       571  09/07/21  30/07/2021  12/07/21  27/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... EXAMINADOR  ID_LVC  GESTANTE   \
0          130260       137    4274.0  ...     4315.0        2         6   
1          130260       137    2118.0  ...     2118.0        2         6   
2          130260       137    4274.0  ...     4315.0        1         5   
3          130260       137    4060.0  ...       17.0        2         6   
4          130260        35    3601.0  ...     3601.0        2         6   
...           ...       ...       ...  ...        ...      ...       ...   
1006354    140002       249     320.0  ...      320.0        2         6   
1006355    160050       128  339967.0  ...    95384.0        2         6   
1006357    130353         9      31.0  ...       31.0        1         6   
1006358    130380        48     666.0  ...      135.0        2         6   
1006359    130250        68      19.0  ...       19.0        1         6   

        AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT  VIVAX FALCIPARUM  NIV_ESCO  \
0               NaN           NaN            NaN    2.0        2.0       5.0   
1               NaN           NaN            NaN    2.0        2.0       4.0   
2               NaN           NaN            NaN    1.0        2.0       6.0   
3               NaN           NaN            NaN    2.0        2.0       3.0   
4               NaN           NaN            NaN    2.0        2.0       5.0   
...             ...           ...            ...    ...        ...       ...   
1006354         NaN           NaN            NaN    2.0        2.0       NaN   
1006355         NaN           NaN            NaN    2.0        2.0       4.0   
1006357         NaN           NaN            NaN    1.0        2.0       3.0   
1006358         NaN           NaN            NaN    2.0        2.0       5.0   
1006359         NaN           NaN            NaN    1.0        2.0       4.0   

         YEAR_NOTIF  
0              2017  
1              2017  
2              2017  
3              2017  
4              2017  
...             ...  
1006354        2021  
1006355        2021  
1006357        2021  
1006358        2021  
1006359        2021  

[884532 rows x 49 columns]

In [22]:
corrected_ufs_dfs_manaus = corrected_ufs_df[corrected_ufs_df.get('MUN_RESI')== 130260]
corrected_ufs_dfs_manaus

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1005991      2653  25/09/21  29/09/2021  28/09/21  38/2021         1       AM   
1006093       181  03/11/21  04/11/2021  04/11/21  44/2021         1       AM   
1006111    187216  08/10/21  10/10/2021  08/10/21  40/2021         1       AM   
1006159       148  19/10/21  21/10/2021  21/10/21  42/2021         1       AM   
1006319        67  02/08/21  03/08/2021  03/08/21  31/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... EXAMINADOR  ID_LVC  GESTANTE   \
0          130260       137    4274.0  ...     4315.0        2         6   
1          130260       137    2118.0  ...     2118.0        2         6   
2          130260       137    4274.0  ...     4315.0        1         5   
3          130260       137    4060.0  ...       17.0        2         6   
4          130260        35    3601.0  ...     3601.0        2         6   
...           ...       ...       ...  ...        ...      ...       ...   
1005991    130260       411    2573.0  ...     2573.0        2         6   
1006093    130260       385    4799.0  ...     3208.0        2         5   
1006111    130260        70     338.0  ...     2056.0        2         6   
1006159    130260        42    3168.0  ...      593.0        2         6   
1006319    130260        46    2712.0  ...     3278.0        2         5   

        AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT  VIVAX FALCIPARUM  NIV_ESCO  \
0               NaN           NaN            NaN    2.0        2.0       5.0   
1               NaN           NaN            NaN    2.0        2.0       4.0   
2               NaN           NaN            NaN    1.0        2.0       6.0   
3               NaN           NaN            NaN    2.0        2.0       3.0   
4               NaN           NaN            NaN    2.0        2.0       5.0   
...             ...           ...            ...    ...        ...       ...   
1005991         NaN           NaN            NaN    2.0        2.0       5.0   
1006093         2.0           NaN            NaN    2.0        2.0       6.0   
1006111         NaN           NaN            NaN    2.0        2.0       5.0   
1006159         NaN           NaN            NaN    2.0        2.0       3.0   
1006319         2.0           NaN            NaN    2.0        2.0       5.0   

         YEAR_NOTIF  
0              2017  
1              2017  
2              2017  
3              2017  
4              2017  
...             ...  
1005991        2021  
1006093        2021  
1006111        2021  
1006159        2021  
1006319        2021  

[52079 rows x 49 columns]

Getting cases in Manaus by year

In [23]:
tot_cases_manaus_by_year = corrected_ufs_df.groupby(['MUN_RESI', 'YEAR_NOTIF']).size().reset_index(name='TOTAL_CASES').sort_values(by='TOTAL_CASES', ascending=False)
tot_cases_manaus_by_year = tot_cases_manaus_by_year.sort_values(by=['YEAR_NOTIF', 'MUN_RESI'], ascending = True)
tot_cases_manaus_by_year['MUN_RESI'] = tot_cases_manaus_by_year['MUN_RESI'].astype(int)
tot_cases_manaus_by_year['MUN_RESI'] = tot_cases_manaus_by_year['MUN_RESI'].astype(str)
tot_cases_manaus_by_year = tot_cases_manaus_by_year[tot_cases_manaus_by_year.get('MUN_RESI') == '130260']
tot_cases_manaus_by_year

MUN_RESI  YEAR_NOTIF  TOTAL_CASES
531   130260        2017        16042
532   130260        2018        12618
533   130260        2019         9720
534   130260        2020         7501
535   130260        2021         6198

In [24]:
full_pop_df['MUNIC_RES'] = full_pop_df['MUNIC_RES'].astype(str)
full_pop_df

MUNIC_RES  YEAR  POPULATION
77926    1100015  2017       25437
77927    1100023  2017      107345
77928    1100031  2017        6224
77929    1100049  2017       88507
77930    1100056  2017       17934
...          ...   ...         ...
105771   5222005  2021       14088
105772   5222054  2021        9002
105773   5222203  2021        6451
105774   5222302  2021        5941
105775   5300108  2021     3094325

[27850 rows x 3 columns]

Corrects IBGE code to limit to 6 digits

In [25]:
for i in np.arange(0, len(full_pop_df['MUNIC_RES'])):
    if len(full_pop_df['MUNIC_RES'].iloc[i]) > 6:
        full_pop_df['MUNIC_RES'].iloc[i] = full_pop_df['MUNIC_RES'].iloc[i][:-1]
    
full_pop_df

C:\Users\ilana\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)


MUNIC_RES  YEAR  POPULATION
77926     110001  2017       25437
77927     110002  2017      107345
77928     110003  2017        6224
77929     110004  2017       88507
77930     110005  2017       17934
...          ...   ...         ...
105771    522200  2021       14088
105772    522205  2021        9002
105773    522220  2021        6451
105774    522230  2021        5941
105775    530010  2021     3094325

[27850 rows x 3 columns]

In [26]:
full_pop_manaus_df = full_pop_df
full_pop_manaus_df['MUNIC_RES'] = full_pop_manaus_df['MUNIC_RES'].astype(str)
full_pop_manaus_df = full_pop_manaus_df[full_pop_manaus_df.get('MUNIC_RES') == '130260']
full_pop_manaus_df

MUNIC_RES  YEAR  POPULATION
78037     130260  2017     2130264
83607     130260  2018     2145444
89177     130260  2019     2182763
94747     130260  2020     2219580
100317    130260  2021     2255903

In [27]:
tot_cases_manaus_by_year = tot_cases_manaus_by_year.rename(columns={"YEAR_NOTIF": "YEAR"})
full_pop_manaus_df = full_pop_manaus_df.rename(columns={"MUNIC_RES": "MUN_RESI"})

#### Gets number of cases and population per municipality per year

In [28]:
cases_per_year_df = pd.merge(tot_cases_manaus_by_year, full_pop_manaus_df, on=['MUN_RESI', 'YEAR'], how='inner')
cases_per_year_df = cases_per_year_df[['MUN_RESI', 'YEAR', 'TOTAL_CASES', 'POPULATION']]
cases_per_year_df

MUN_RESI  YEAR  TOTAL_CASES  POPULATION
0   130260  2017        16042     2130264
1   130260  2018        12618     2145444
2   130260  2019         9720     2182763
3   130260  2020         7501     2219580
4   130260  2021         6198     2255903

#### Gets annual incidence per municipality per year

In [29]:
annual_incidence_df = cases_per_year_df

annual_incidence_df['CASES_PER_100K'] = (annual_incidence_df['TOTAL_CASES'].astype(int) / annual_incidence_df['POPULATION'].astype(int)) * 100000
annual_incidence_df = annual_incidence_df.sort_values(by='CASES_PER_100K', ascending=False)
annual_incidence_df

MUN_RESI  YEAR  TOTAL_CASES  POPULATION  CASES_PER_100K
0   130260  2017        16042     2130264      753.052204
1   130260  2018        12618     2145444      588.130009
2   130260  2019         9720     2182763      445.307163
3   130260  2020         7501     2219580      337.946819
4   130260  2021         6198     2255903      274.745856

#### Gets total number of cases in each municipality from 2003 to 2023

In [30]:
tot_cases_manaus = tot_cases_brazil[tot_cases_brazil.get('MUN_RESI') == '130260']
tot_cases_manaus

MUN_RESI  TOTAL_CASES
111   130260        52079

In [31]:
full_cases_df = pd.merge(tot_cases_manaus, full_pop_manaus_df, on=['MUN_RESI'], how='inner')
full_cases_df

MUN_RESI  TOTAL_CASES  YEAR  POPULATION
0   130260        52079  2017     2130264
1   130260        52079  2018     2145444
2   130260        52079  2019     2182763
3   130260        52079  2020     2219580
4   130260        52079  2021     2255903

#### Mean population of Manaus from 2017 to 2021

In [32]:
mean_population_df = full_cases_df.groupby('MUN_RESI').agg({
    'POPULATION': 'mean',
    'TOTAL_CASES': 'first'
}).reset_index()

mean_population_df = mean_population_df.rename(columns={"POPULATION": "MEAN_POPULATION"})
mean_population_df

MUN_RESI  MEAN_POPULATION  TOTAL_CASES
0   130260        2186790.8        52079

#### Mean incidence of cases in Manaus from 2017 to 2021

In [33]:
mean_incidence_df = mean_population_df

mean_incidence_df['CASES_PER_100K'] = (mean_incidence_df['TOTAL_CASES'].astype(int) / mean_incidence_df['MEAN_POPULATION'].astype(int)) * 100000
mean_incidence_df = mean_incidence_df.sort_values(by='CASES_PER_100K', ascending=False)
mean_incidence_df

MUN_RESI  MEAN_POPULATION  TOTAL_CASES  CASES_PER_100K
0   130260        2186790.8        52079     2381.527261

In [34]:
manaus_cases_per_year_df = cases_per_year_df
manaus_cases_per_year_df

MUN_RESI  YEAR  TOTAL_CASES  POPULATION  CASES_PER_100K
0   130260  2017        16042     2130264      753.052204
1   130260  2018        12618     2145444      588.130009
2   130260  2019         9720     2182763      445.307163
3   130260  2020         7501     2219580      337.946819
4   130260  2021         6198     2255903      274.745856

In [35]:
manaus_rural_pop_df = manaus_cases_per_year_df.assign(RURAL_POP = manaus_cases_per_year_df.get('POPULATION')*0.45/100)
manaus_rural_pop_df['RURAL_POP'] = manaus_rural_pop_df['RURAL_POP'].round()
manaus_rural_pop_df = manaus_rural_pop_df.assign(RURAL_CASES = manaus_rural_pop_df.get('TOTAL_CASES')*35.75/100)

manaus_rural_pop_df

MUN_RESI  YEAR  TOTAL_CASES  POPULATION  CASES_PER_100K  RURAL_POP  \
0   130260  2017        16042     2130264      753.052204     9586.0   
1   130260  2018        12618     2145444      588.130009     9654.0   
2   130260  2019         9720     2182763      445.307163     9822.0   
3   130260  2020         7501     2219580      337.946819     9988.0   
4   130260  2021         6198     2255903      274.745856    10152.0   

   RURAL_CASES  
0    5735.0150  
1    4510.9350  
2    3474.9000  
3    2681.6075  
4    2215.7850

In [36]:
manaus_rural_pop_df['RURAL_INCIDENCE'] = (manaus_rural_pop_df['RURAL_CASES'].astype(int) / manaus_rural_pop_df['RURAL_POP'].astype(int)) * 100000
#manaus_rural_pop_df = manaus_rural_pop_df.sort_values(by='CASES_PER_100K', ascending=False)
manaus_rural_pop_df

MUN_RESI  YEAR  TOTAL_CASES  POPULATION  CASES_PER_100K  RURAL_POP  \
0   130260  2017        16042     2130264      753.052204     9586.0   
1   130260  2018        12618     2145444      588.130009     9654.0   
2   130260  2019         9720     2182763      445.307163     9822.0   
3   130260  2020         7501     2219580      337.946819     9988.0   
4   130260  2021         6198     2255903      274.745856    10152.0   

   RURAL_CASES  RURAL_INCIDENCE  
0    5735.0150     59826.830795  
1    4510.9350     46716.386990  
2    3474.9000     35369.578497  
3    2681.6075     26842.210653  
4    2215.7850     21818.360914

In [37]:
manaus_pop_difference_df = manaus_rural_pop_df

manaus_pop_difference_df['POPULATION_DIFF'] = manaus_pop_difference_df['POPULATION'] - manaus_pop_difference_df['POPULATION'].shift(1)
manaus_pop_difference_df['POPULATION_DIFF'].iloc[0] = 0

manaus_pop_difference_df['RURAL_POP_DIFF'] = manaus_pop_difference_df['RURAL_POP'] - manaus_pop_difference_df['RURAL_POP'].shift(1)
manaus_pop_difference_df['RURAL_POP_DIFF'].iloc[0] = 0

manaus_pop_difference_df

C:\Users\ilana\anaconda3\lib\site-packages\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)


MUN_RESI  YEAR  TOTAL_CASES  POPULATION  CASES_PER_100K  RURAL_POP  \
0   130260  2017        16042     2130264      753.052204     9586.0   
1   130260  2018        12618     2145444      588.130009     9654.0   
2   130260  2019         9720     2182763      445.307163     9822.0   
3   130260  2020         7501     2219580      337.946819     9988.0   
4   130260  2021         6198     2255903      274.745856    10152.0   

   RURAL_CASES  RURAL_INCIDENCE  POPULATION_DIFF  RURAL_POP_DIFF  
0    5735.0150     59826.830795              0.0             0.0  
1    4510.9350     46716.386990          15180.0            68.0  
2    3474.9000     35369.578497          37319.0           168.0  
3    2681.6075     26842.210653          36817.0           166.0  
4    2215.7850     21818.360914          36323.0           164.0

Total increase in population between 2017 and 2021:

In [38]:
manaus_pop_difference_df.get('RURAL_POP').iloc[-1]-manaus_pop_difference_df.get('RURAL_POP').iloc[0]

566.0

Approximating a birth rate between first and last years:

In [39]:
(manaus_pop_difference_df.get('RURAL_POP').iloc[-1]-manaus_pop_difference_df.get('RURAL_POP').iloc[0])/(len(manaus_pop_difference_df)-1)

141.5

About 142 births per year from 2017 to 2021

Getting a mean rural population

In [40]:
mean_rural_pop = manaus_pop_difference_df.loc[0:4, 'RURAL_POP'].mean()
mean_rural_pop

9840.4

Around 9840 people living in the rural areas of Manaus

In [41]:
manaus_rural_births_per_day = (manaus_pop_difference_df.get('RURAL_POP').iloc[-1]-manaus_pop_difference_df.get('RURAL_POP').iloc[0])/(len(manaus_pop_difference_df)-1)/365
manaus_rural_births_per_day

0.38767123287671235

Around 0.388 births per day

In [42]:
manaus_rural_births_per_person_day = manaus_rural_births_per_day/mean_rural_pop
manaus_rural_births_per_person_day

3.939588155732616e-05

Around 0.0000394 births per person per year ($\mu_H$)

#### Searching for monthly cases of malaria in residents of Manaus

In [43]:
manaus_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0            2990  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
1             421  31/01/17  03/02/2017  03/02/17  05/2017         1       AM   
2            2991  10/09/17  13/09/2017  13/09/17  37/2017         1       AM   
3            5196  09/12/17  14/12/2017  14/12/17  49/2017         1       AM   
4            7891  24/08/17  25/08/2017  25/08/17  34/2017         1       AM   
...           ...       ...         ...       ...      ...       ...      ...   
1005991      2653  25/09/21  29/09/2021  28/09/21  38/2021         1       AM   
1006093       181  03/11/21  04/11/2021  04/11/21  44/2021         1       AM   
1006111    187216  08/10/21  10/10/2021  08/10/21  40/2021         1       AM   
1006159       148  19/10/21  21/10/2021  21/10/21  42/2021         1       AM   
1006319        67  02/08/21  03/08/2021  03/08/21  31/2021         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... EXAMINADOR  ID_LVC  GESTANTE   \
0          130260       137    4274.0  ...     4315.0        2         6   
1          130260       137    2118.0  ...     2118.0        2         6   
2          130260       137    4274.0  ...     4315.0        1         5   
3          130260       137    4060.0  ...       17.0        2         6   
4          130260        35    3601.0  ...     3601.0        2         6   
...           ...       ...       ...  ...        ...      ...       ...   
1005991    130260       411    2573.0  ...     2573.0        2         6   
1006093    130260       385    4799.0  ...     3208.0        2         5   
1006111    130260        70     338.0  ...     2056.0        2         6   
1006159    130260        42    3168.0  ...      593.0        2         6   
1006319    130260        46    2712.0  ...     3278.0        2         5   

        AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT  VIVAX FALCIPARUM  NIV_ESCO  \
0               NaN           NaN            NaN    2.0        2.0       5.0   
1               NaN           NaN            NaN    2.0        2.0       4.0   
2               NaN           NaN            NaN    1.0        2.0       6.0   
3               NaN           NaN            NaN    2.0        2.0       3.0   
4               NaN           NaN            NaN    2.0        2.0       5.0   
...             ...           ...            ...    ...        ...       ...   
1005991         NaN           NaN            NaN    2.0        2.0       5.0   
1006093         2.0           NaN            NaN    2.0        2.0       6.0   
1006111         NaN           NaN            NaN    2.0        2.0       5.0   
1006159         NaN           NaN            NaN    2.0        2.0       3.0   
1006319         2.0           NaN            NaN    2.0        2.0       5.0   

         YEAR_NOTIF  
0              2017  
1              2017  
2              2017  
3              2017  
4              2017  
...             ...  
1005991        2021  
1006093        2021  
1006111        2021  
1006159        2021  
1006319        2021  

[52079 rows x 49 columns]

In [44]:
manaus_resid.columns

Index(['COD_NOTI', 'DT_NOTIF', 'DT_ENVLO', 'DT_DIGIT', 'SEM_NOTI', 'TIPO_LAM',
       'UF_NOTIF', 'MUN_NOTI', 'COD_UNIN', 'COD_AGEN', 'DT_NASCI', 'ID_PACIE',
       'ID_DIMEA', 'SEXO', 'RACA', 'PESO', 'PAIS_RES', 'UF_RESID', 'MUN_RESI',
       'LOC_RESI', 'SINTOMAS', 'DT_SINTO', 'COD_OCUP', 'UF_INFEC', 'PAIS_INF',
       'MUN_INFE', 'LOC_INFE', 'DT_EXAME', 'RES_EXAM', 'QTD_PARA', 'QTD_CRUZ',
       'DT_TRATA', 'ESQUEMA', 'CO_FORMA_SANGUINEA',
       'CO_FORMA_TECIDUAL_GAMETOCITO', 'HEMOPARASI', 'EXAME ',
       'NU_ATIVIDADE_G6PD', 'NU_HEMOGLOBINA_TOTAL', 'EXAMINADOR', 'ID_LVC ',
       'GESTANTE ', 'AMAMENTANDO', 'TEMP_AMAMENT', 'DIMEA_AMAMENT', 'VIVAX',
       'FALCIPARUM', 'NIV_ESCO', 'YEAR_NOTIF'],
      dtype='object')

In [45]:
manaus_resid_dropped_columns = manaus_resid.drop(columns=['DT_TRATA', 'ESQUEMA', 'CO_FORMA_SANGUINEA', 
                                'CO_FORMA_TECIDUAL_GAMETOCITO', 'EXAME ', 'NU_ATIVIDADE_G6PD', 
                                'NU_HEMOGLOBINA_TOTAL', 'EXAMINADOR', 'ID_LVC ', 'GESTANTE ', 
                                'AMAMENTANDO', 'TEMP_AMAMENT', 'DIMEA_AMAMENT', 'NIV_ESCO', 'FALCIPARUM',
                                'QTD_PARA', 'QTD_CRUZ', 'SEXO', 'RACA', 'PESO', 'COD_AGEN', 'DT_NASCI', 'ID_PACIE',
                                'ID_DIMEA', 'COD_UNIN', 'COD_NOTI', 'TIPO_LAM', 'DT_ENVLO', 'HEMOPARASI', 
                                'PAIS_INF', 'PAIS_RES', 'DT_DIGIT'])
manaus_resid_dropped_columns

DT_NOTIF SEM_NOTI UF_NOTIF  MUN_NOTI UF_RESID  MUN_RESI  LOC_RESI  \
0        10/09/17  37/2017       AM    130260       AM  130260.0     267.0   
1        31/01/17  05/2017       AM    130260       AM  130260.0     564.0   
2        10/09/17  37/2017       AM    130260       AM  130260.0     267.0   
3        09/12/17  49/2017       AM    130260       AM  130260.0     564.0   
4        24/08/17  34/2017       AM    130260       AM  130260.0     565.0   
...           ...      ...      ...       ...      ...       ...       ...   
1005991  25/09/21  38/2021       AM    130260       AM  130260.0     577.0   
1006093  03/11/21  44/2021       AM    130260       AM  130260.0     483.0   
1006111  08/10/21  40/2021       AM    130260       AM  130260.0     395.0   
1006159  19/10/21  42/2021       AM    130260       AM  130260.0     218.0   
1006319  02/08/21  31/2021       AM    130260       AM  130260.0     490.0   

         SINTOMAS  DT_SINTO  COD_OCUP  UF_INFEC  MUN_INFE  LOC_INFE  DT_EXAME  \
0               1  02/09/17      11.0      13.0  130260.0     564.0  10/09/17   
1               1  28/01/17      11.0      13.0  130260.0     564.0  31/01/17   
2               1  02/09/17      11.0      13.0  130260.0     564.0  10/09/17   
3               1  07/12/17      11.0      13.0  130260.0     564.0  09/12/17   
4               1  20/08/17      11.0      13.0  130260.0     565.0  24/08/17   
...           ...       ...       ...       ...       ...       ...       ...   
1005991         1  20/09/21      11.0       NaN       NaN       NaN  25/09/21   
1006093         1  01/11/21      11.0      14.0  140010.0       NaN  03/11/21   
1006111         1  01/08/21      11.0       NaN       NaN       NaN  08/10/21   
1006159         1  19/10/21      11.0      15.0  150140.0       NaN  19/10/21   
1006319         1  29/07/21       5.0       NaN       NaN       NaN  02/08/21   

         RES_EXAM  VIVAX  YEAR_NOTIF  
0               4    2.0        2017  
1               4    2.0        2017  
2               4    1.0        2017  
3               4    2.0        2017  
4               4    2.0        2017  
...           ...    ...         ...  
1005991         4    2.0        2021  
1006093         4    2.0        2021  
1006111         4    2.0        2021  
1006159         4    2.0        2021  
1006319         4    2.0        2021  

[52079 rows x 17 columns]

In [46]:
manaus_resid_dropped_columns.columns

Index(['DT_NOTIF', 'SEM_NOTI', 'UF_NOTIF', 'MUN_NOTI', 'UF_RESID', 'MUN_RESI',
       'LOC_RESI', 'SINTOMAS', 'DT_SINTO', 'COD_OCUP', 'UF_INFEC', 'MUN_INFE',
       'LOC_INFE', 'DT_EXAME', 'RES_EXAM', 'VIVAX', 'YEAR_NOTIF'],
      dtype='object')

In [47]:
manaus_resid_dropped_columns['DT_NOTIF'] = pd.to_datetime(manaus_resid_dropped_columns['DT_NOTIF'], format='%d/%m/%y')

manaus_resid_dropped_columns['MONTH_NOTIF'] = manaus_resid_dropped_columns['DT_NOTIF'].dt.strftime('%m/%Y')

manaus_resid_dropped_columns

DT_NOTIF SEM_NOTI UF_NOTIF  MUN_NOTI UF_RESID  MUN_RESI  LOC_RESI  \
0       2017-09-10  37/2017       AM    130260       AM  130260.0     267.0   
1       2017-01-31  05/2017       AM    130260       AM  130260.0     564.0   
2       2017-09-10  37/2017       AM    130260       AM  130260.0     267.0   
3       2017-12-09  49/2017       AM    130260       AM  130260.0     564.0   
4       2017-08-24  34/2017       AM    130260       AM  130260.0     565.0   
...            ...      ...      ...       ...      ...       ...       ...   
1005991 2021-09-25  38/2021       AM    130260       AM  130260.0     577.0   
1006093 2021-11-03  44/2021       AM    130260       AM  130260.0     483.0   
1006111 2021-10-08  40/2021       AM    130260       AM  130260.0     395.0   
1006159 2021-10-19  42/2021       AM    130260       AM  130260.0     218.0   
1006319 2021-08-02  31/2021       AM    130260       AM  130260.0     490.0   

         SINTOMAS  DT_SINTO  COD_OCUP  UF_INFEC  MUN_INFE  LOC_INFE  DT_EXAME  \
0               1  02/09/17      11.0      13.0  130260.0     564.0  10/09/17   
1               1  28/01/17      11.0      13.0  130260.0     564.0  31/01/17   
2               1  02/09/17      11.0      13.0  130260.0     564.0  10/09/17   
3               1  07/12/17      11.0      13.0  130260.0     564.0  09/12/17   
4               1  20/08/17      11.0      13.0  130260.0     565.0  24/08/17   
...           ...       ...       ...       ...       ...       ...       ...   
1005991         1  20/09/21      11.0       NaN       NaN       NaN  25/09/21   
1006093         1  01/11/21      11.0      14.0  140010.0       NaN  03/11/21   
1006111         1  01/08/21      11.0       NaN       NaN       NaN  08/10/21   
1006159         1  19/10/21      11.0      15.0  150140.0       NaN  19/10/21   
1006319         1  29/07/21       5.0       NaN       NaN       NaN  02/08/21   

         RES_EXAM  VIVAX  YEAR_NOTIF MONTH_NOTIF  
0               4    2.0        2017     09/2017  
1               4    2.0        2017     01/2017  
2               4    1.0        2017     09/2017  
3               4    2.0        2017     12/2017  
4               4    2.0        2017     08/2017  
...           ...    ...         ...         ...  
1005991         4    2.0        2021     09/2021  
1006093         4    2.0        2021     11/2021  
1006111         4    2.0        2021     10/2021  
1006159         4    2.0        2021     10/2021  
1006319         4    2.0        2021     08/2021  

[52079 rows x 18 columns]

In [48]:
manaus_resid_dropped_columns = manaus_resid_dropped_columns.drop(columns=['UF_INFEC', 'MUN_INFE', 'DT_EXAME'])

# manaus_resid_dropped_columns = manaus_resid_dropped_columns.assign(RURAL_POP = manaus_cases_per_year_df.get('POPULATION')*0.45/100)
# manaus_rural_pop_df['RURAL_POP'] = manaus_rural_pop_df['RURAL_POP'].round()
# manaus_rural_pop_df = manaus_rural_pop_df.assign(RURAL_CASES = manaus_rural_pop_df.get('TOTAL_CASES')*35.75/100)

manaus_resid_dropped_columns

DT_NOTIF SEM_NOTI UF_NOTIF  MUN_NOTI UF_RESID  MUN_RESI  LOC_RESI  \
0       2017-09-10  37/2017       AM    130260       AM  130260.0     267.0   
1       2017-01-31  05/2017       AM    130260       AM  130260.0     564.0   
2       2017-09-10  37/2017       AM    130260       AM  130260.0     267.0   
3       2017-12-09  49/2017       AM    130260       AM  130260.0     564.0   
4       2017-08-24  34/2017       AM    130260       AM  130260.0     565.0   
...            ...      ...      ...       ...      ...       ...       ...   
1005991 2021-09-25  38/2021       AM    130260       AM  130260.0     577.0   
1006093 2021-11-03  44/2021       AM    130260       AM  130260.0     483.0   
1006111 2021-10-08  40/2021       AM    130260       AM  130260.0     395.0   
1006159 2021-10-19  42/2021       AM    130260       AM  130260.0     218.0   
1006319 2021-08-02  31/2021       AM    130260       AM  130260.0     490.0   

         SINTOMAS  DT_SINTO  COD_OCUP  LOC_INFE  RES_EXAM  VIVAX  YEAR_NOTIF  \
0               1  02/09/17      11.0     564.0         4    2.0        2017   
1               1  28/01/17      11.0     564.0         4    2.0        2017   
2               1  02/09/17      11.0     564.0         4    1.0        2017   
3               1  07/12/17      11.0     564.0         4    2.0        2017   
4               1  20/08/17      11.0     565.0         4    2.0        2017   
...           ...       ...       ...       ...       ...    ...         ...   
1005991         1  20/09/21      11.0       NaN         4    2.0        2021   
1006093         1  01/11/21      11.0       NaN         4    2.0        2021   
1006111         1  01/08/21      11.0       NaN         4    2.0        2021   
1006159         1  19/10/21      11.0       NaN         4    2.0        2021   
1006319         1  29/07/21       5.0       NaN         4    2.0        2021   

        MONTH_NOTIF  
0           09/2017  
1           01/2017  
2           09/2017  
3           12/2017  
4           08/2017  
...             ...  
1005991     09/2021  
1006093     11/2021  
1006111     10/2021  
1006159     10/2021  
1006319     08/2021  

[52079 rows x 15 columns]

In [49]:
cases_manaus_by_month = manaus_resid_dropped_columns[['MONTH_NOTIF']].groupby(['MONTH_NOTIF']).size().reset_index(name='TOTAL_CASES')#.sort_values(by='TOTAL_CASES', ascending=False)
cases_manaus_by_month['MONTH_NOTIF'] = cases_manaus_by_month['MONTH_NOTIF']#.astype(int)
cases_manaus_by_month

MONTH_NOTIF  TOTAL_CASES
0      01/2017          983
1      01/2018         1618
2      01/2019          962
3      01/2020          850
4      01/2021          408
5      02/2017          616
6      02/2018         1259
7      02/2019          604
8      02/2020          617
9      02/2021          309
10     03/2017          663
11     03/2018         1069
12     03/2019          501
13     03/2020          592
14     03/2021          304
15     04/2017          537
16     04/2018          880
17     04/2019          593
18     04/2020          329
19     04/2021          289
20     05/2017          944
21     05/2018          847
22     05/2019          537
23     05/2020          394
24     05/2021          284
25     06/2017         1166
26     06/2018          727
27     06/2019          568
28     06/2020          519
29     06/2021          420
30     07/2017         1701
31     07/2018          964
32     07/2019          912
33     07/2020          670
34     07/2021          603
35     08/2017         2162
36     08/2018         1626
37     08/2019         1099
38     08/2020          696
39     08/2021          682
40     09/2017         2291
41     09/2018         1235
42     09/2019         1170
43     09/2020          684
44     09/2021          972
45     10/2017         1864
46     10/2018         1013
47     10/2019          982
48     10/2020          747
49     10/2021          807
50     11/2017         1534
51     11/2018          704
52     11/2019          821
53     11/2020          737
54     11/2021          655
55     12/2017         1581
56     12/2018          676
57     12/2019          971
58     12/2020          666
59     12/2021          465

In [50]:
cases_manaus_by_month['MONTH_NOTIF'] = pd.to_datetime(cases_manaus_by_month['MONTH_NOTIF'], format='%m/%Y')
cases_manaus_by_month = cases_manaus_by_month.sort_values(by='MONTH_NOTIF')
cases_manaus_by_month = cases_manaus_by_month.reset_index(drop=True)
cases_manaus_by_month

MONTH_NOTIF  TOTAL_CASES
0   2017-01-01          983
1   2017-02-01          616
2   2017-03-01          663
3   2017-04-01          537
4   2017-05-01          944
5   2017-06-01         1166
6   2017-07-01         1701
7   2017-08-01         2162
8   2017-09-01         2291
9   2017-10-01         1864
10  2017-11-01         1534
11  2017-12-01         1581
12  2018-01-01         1618
13  2018-02-01         1259
14  2018-03-01         1069
15  2018-04-01          880
16  2018-05-01          847
17  2018-06-01          727
18  2018-07-01          964
19  2018-08-01         1626
20  2018-09-01         1235
21  2018-10-01         1013
22  2018-11-01          704
23  2018-12-01          676
24  2019-01-01          962
25  2019-02-01          604
26  2019-03-01          501
27  2019-04-01          593
28  2019-05-01          537
29  2019-06-01          568
30  2019-07-01          912
31  2019-08-01         1099
32  2019-09-01         1170
33  2019-10-01          982
34  2019-11-01          821
35  2019-12-01          971
36  2020-01-01          850
37  2020-02-01          617
38  2020-03-01          592
39  2020-04-01          329
40  2020-05-01          394
41  2020-06-01          519
42  2020-07-01          670
43  2020-08-01          696
44  2020-09-01          684
45  2020-10-01          747
46  2020-11-01          737
47  2020-12-01          666
48  2021-01-01          408
49  2021-02-01          309
50  2021-03-01          304
51  2021-04-01          289
52  2021-05-01          284
53  2021-06-01          420
54  2021-07-01          603
55  2021-08-01          682
56  2021-09-01          972
57  2021-10-01          807
58  2021-11-01          655
59  2021-12-01          465

In [51]:
cases_manaus_by_month['TOTAL_CASES'].sum()

52079

In [52]:
cases_manaus_by_month = cases_manaus_by_month.assign(RURAL_CASES = cases_manaus_by_month.get('TOTAL_CASES')*35.75/100)
cases_manaus_by_month

MONTH_NOTIF  TOTAL_CASES  RURAL_CASES
0   2017-01-01          983     351.4225
1   2017-02-01          616     220.2200
2   2017-03-01          663     237.0225
3   2017-04-01          537     191.9775
4   2017-05-01          944     337.4800
5   2017-06-01         1166     416.8450
6   2017-07-01         1701     608.1075
7   2017-08-01         2162     772.9150
8   2017-09-01         2291     819.0325
9   2017-10-01         1864     666.3800
10  2017-11-01         1534     548.4050
11  2017-12-01         1581     565.2075
12  2018-01-01         1618     578.4350
13  2018-02-01         1259     450.0925
14  2018-03-01         1069     382.1675
15  2018-04-01          880     314.6000
16  2018-05-01          847     302.8025
17  2018-06-01          727     259.9025
18  2018-07-01          964     344.6300
19  2018-08-01         1626     581.2950
20  2018-09-01         1235     441.5125
21  2018-10-01         1013     362.1475
22  2018-11-01          704     251.6800
23  2018-12-01          676     241.6700
24  2019-01-01          962     343.9150
25  2019-02-01          604     215.9300
26  2019-03-01          501     179.1075
27  2019-04-01          593     211.9975
28  2019-05-01          537     191.9775
29  2019-06-01          568     203.0600
30  2019-07-01          912     326.0400
31  2019-08-01         1099     392.8925
32  2019-09-01         1170     418.2750
33  2019-10-01          982     351.0650
34  2019-11-01          821     293.5075
35  2019-12-01          971     347.1325
36  2020-01-01          850     303.8750
37  2020-02-01          617     220.5775
38  2020-03-01          592     211.6400
39  2020-04-01          329     117.6175
40  2020-05-01          394     140.8550
41  2020-06-01          519     185.5425
42  2020-07-01          670     239.5250
43  2020-08-01          696     248.8200
44  2020-09-01          684     244.5300
45  2020-10-01          747     267.0525
46  2020-11-01          737     263.4775
47  2020-12-01          666     238.0950
48  2021-01-01          408     145.8600
49  2021-02-01          309     110.4675
50  2021-03-01          304     108.6800
51  2021-04-01          289     103.3175
52  2021-05-01          284     101.5300
53  2021-06-01          420     150.1500
54  2021-07-01          603     215.5725
55  2021-08-01          682     243.8150
56  2021-09-01          972     347.4900
57  2021-10-01          807     288.5025
58  2021-11-01          655     234.1625
59  2021-12-01          465     166.2375

In [53]:
cases_2017 = cases_manaus_by_month[cases_manaus_by_month['MONTH_NOTIF'].dt.year == 2017]
cases_2017.get('TOTAL_CASES').sum()

16042

In [54]:
cases_2018 = cases_manaus_by_month[cases_manaus_by_month['MONTH_NOTIF'].dt.year == 2018]
cases_2018.get('TOTAL_CASES').sum()

12618

In [55]:
cases_2019 = cases_manaus_by_month[cases_manaus_by_month['MONTH_NOTIF'].dt.year == 2019]
cases_2019.get('TOTAL_CASES').sum()

9720

In [56]:
cases_2020 = cases_manaus_by_month[cases_manaus_by_month['MONTH_NOTIF'].dt.year == 2020]
cases_2020.get('TOTAL_CASES').sum()

7501

In [57]:
cases_2021 = cases_manaus_by_month[cases_manaus_by_month['MONTH_NOTIF'].dt.year == 2021]
cases_2021.get('TOTAL_CASES').sum()

6198